codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2HeadersDecoder.java

```diff
@@ -50,10 +56,23 @@ public Http2HeaderTable headerTable() {
     }
 
     @Override
+    public int maxHeaderSize() {
+        return maxHeaderSize;
+    }
+
+    @Override
     public Configuration configuration() {
         return this;
     }
 
+    /**
+     * Respond to headers block resulting in the maximum header size being exceeded.
+     * @throws Http2Exception If we can not recover from the truncation.
+     */
+    protected void maxHeaderSizeExceeded() throws Http2Exception {
```

Comment:
The intention was to make it extensible.  The default behavior is to `throw`, but some implementations may want to carry on with what they have?